core/src/main/java/com/sequenceiq/cloudbreak/service/RotateSaltPasswordService.java

```diff
@@ -62,14 +77,47 @@ private boolean isChangeSaltuserPasswordSupported(StackDto stack) {
 
     public void rotateSaltPassword(StackDto stack) throws CloudbreakOrchestratorException {
         validateRotateSaltPassword(stack);
-        SecurityConfig securityConfig = securityConfigService.getOneByStackId(stack.getId());
+        SecurityConfig securityConfig = stack.getSecurityConfig();
         String oldPassword = securityConfig.getSaltSecurityConfig().getSaltPassword();
         String newPassword = PasswordUtil.generatePassword();
         List<GatewayConfig> allGatewayConfig = gatewayConfigService.getAllGatewayConfigs(stack);
         hostOrchestrator.changePassword(allGatewayConfig, newPassword, oldPassword);
         securityConfigService.changeSaltPassword(securityConfig, newPassword);
     }
 
+    public void rotateSaltPasswordFallback(StackDto stack) throws CloudbreakOrchestratorFailedException {
+        List<GatewayConfig> allGatewayConfig = gatewayConfigService.getAllGatewayConfigs(stack);
+
+        try {
+            Map<String, String> response = hostOrchestrator.runCommandOnHosts(allGatewayConfig, stack.getAllPrimaryGatewayInstanceNodes(),
+                    SALTUSER_DELETE_COMMAND);
+            LOGGER.debug("Saltuser delete command response: {}", response);
+        } catch (CloudbreakOrchestratorFailedException e) {
+            LOGGER.warn("Failed to run saltuser delete command, assuming it is already deleted", e);
+        }
+        String newPassword = PasswordUtil.generatePassword();
+        SecurityConfig securityConfig = stack.getSecurityConfig();
+        securityConfig.getSaltSecurityConfig().setSaltPassword(newPassword);
+        try {
+            InMemoryStateStore.putStack(stack.getId(), PollGroup.POLLABLE);
```

Comment:
we usually do this via `StakcUpdater` by setting the cluster to status, like `UPDATE_IN_PROGRESS`. Is there a specific reason you do this in a different way? What's the stack status during rotation?

Reply:
The status is handled there: https://github.com/hortonworks/cloudbreak/pull/13170/files#diff-dd0e788826b4bec1f164a787f94d3b3d5d82c040af53de7f4ed15eb1d3cb0d74R53

Reply:
in that case why is this call necessary?

Reply:
Seems like you are right, it is not needed, so removed it